src/test/java/springsamurais/toyswapbackend/controller/CommentControllerTest.java

```diff
@@ -0,0 +1,152 @@
+        mockComment.setText("very Nice");
+        mockComment.setCommenter(commenter);
+
+        when(commentsService.getCommentsByListingId(1L)).thenReturn(Arrays.asList(mockComment));
+
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/api/v1/comments/listing/1"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$[0].commentId").value(1L))
+                .andExpect(jsonPath("$[0].text").value("very Nice"))
+                .andExpect(jsonPath("$[0].commenter.id").value(1L));
+
+    }
+
+    @Test
+    void saveComment() throws Exception {
+
+        Member commenter = new Member(1L, "name", "member", "location", null);
+
+        Comment mockComment = new Comment();
+        mockComment.setCommentId(1L);
+        mockComment.setText("very Nice");
+        mockComment.setCommenter(commenter);
+
+        when(commentsService.saveComment(any(Comment.class))).thenReturn(mockComment);
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        this.mockMvc.perform(MockMvcRequestBuilders.post("/api/v1/comments")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{ \"text\": \"very Nice\", \"commenter\": { \"id\": 1 } }"))
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.commentId").value(1L))
+                .andExpect(jsonPath("$.text").value("very Nice"))
+                .andExpect(jsonPath("$.commenter.id").value(1L));
+
+    }
+
+    @Test
+    void deleteCommentByCommentId() throws Exception {
+        when(commentsService.deleteCommentByCommentId(1L)).thenReturn("Comment deleted" );
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/comments/{commentId}", 1L))
+                .andExpect(status().isOk());
+
+        verify(commentsService, times(1)).deleteCommentByCommentId(1L);
+
+    }
+
+    @Test
+    void deleteCommentsByListingId() throws Exception {
+
+       when(commentsService.deleteCommentsByListingId(1L)).thenReturn("Comment deleted");
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/comments/listing/{listingId}", 1L))
+                .andExpect(status().isOk());
+
+        verify(commentsService, times(1)).deleteCommentsByListingId(1L);
+    }
+
+    @Test
+    void deleteCommentsByUserId() throws Exception {
+        // Mocking the service behavior
+        when(commentsService.deleteCommentsByUserId(1L)).thenReturn("Comment deleted");
+
+        // Performing the request and verifying response
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/comments/user/{userId}", 1L))
+                .andExpect(status().isOk());
+
+        // Verify that the service method was called with the correct userId
+        verify(commentsService, times(1)).deleteCommentsByUserId(1L);
+    }
+}
```

Comment:
Just FYI I haven't reviewed testing. Trusting that these are ok and working as intended.